mrp_operations_schedule/model/mrp.py

```diff
@@ -0,0 +1,212 @@
+            cr, uid, ids, properties=properties, context=context)
+        self._reschedule_workorders(cr, uid, ids, context=context)
+        return result
+
+
+class MrpProductionWorkcenterLine(orm.Model):
+
+    _inherit = "mrp.production.workcenter.line"
+
+    def _get_date_end(self, cr, uid, ids, field_name, arg, context=None):
+        """ Finds ending date.
+        @return: Dictionary of values.
+        """
+        ops = self.browse(cr, uid, ids, context=context)
+        date_and_hours_by_cal = []
+        for op in ops:
+            date_start = op.date_start or op.date_planned
+            if date_start:
+                date_and_hours_by_cal.extend(
+                    [(date_start, op.hour, op.workcenter_id.calendar_id.id)])
+        intervals = self.pool['resource.calendar'].interval_get_multi(
+            cr, uid, date_and_hours_by_cal)
+
+        res = {}
+        for op in ops:
+            res[op.id] = False
+            date_start = op.date_start or op.date_planned
+            if date_start:
+                i = intervals.get((date_start, op.hour,
+                                   op.workcenter_id.calendar_id.id))
+                if i:
+                    res[op.id] = i[-1][1].strftime('%Y-%m-%d %H:%M:%S')
+                else:
+                    res[op.id] = date_start
+        return res
+
+    def _get_gantt_dates(self, cr, uid, ids, field_name, arg, context=None):
+        res = {}
+        if context is None:
+            context = {}
+        for workorder in self.browse(cr, uid, ids, context=context):
+            res[workorder.id] = {}
+            if workorder.date_planned:
+                res[workorder.id]['date_start_gantt'] = \
+                    workorder.date_planned
+            if workorder.date_start:
+                res[workorder.id]['date_start_gantt'] = \
+                    workorder.date_start
+            if workorder.date_planned_end:
+                res[workorder.id]['date_end_gantt'] = \
+                    workorder.date_planned_end
+            if workorder.date_finished:
+                res[workorder.id]['date_end_gantt'] = \
+                    workorder.date_finished
+        return res
+
+    _columns = {
+        'date_start_gantt': fields.function(_get_gantt_dates,
+                                            string='Gantt start date',
+                                            type='datetime',
+                                            multi='gantt_dates'),
+        'date_end_gantt': fields.function(_get_gantt_dates,
+                                          string='Gantt end date',
+                                          type='datetime',
+                                          multi='gantt_dates'),
+        'date_planned_end': fields.function(_get_date_end,
+                                            string='End Date',
+                                            type='datetime',
+                                            store=False),
+    }
```

Comment:
I would make date_planned_end a normal field and rename date_planned to
date_planned_start and use that.  There is a date_start and date_end
already.  What do you think?

On Thu, Nov 26, 2015 at 2:25 PM, Jordi Ballester Alomar <
notifications@github.com> wrote:

> In mrp_operations_schedule/model/mrp.py
> <https://github.com/OCA/manufacture/pull/28#discussion_r45977819>:
>
> > +        return res
> > +
> > +    _columns = {
> > +        'date_start_gantt': fields.function(_get_gantt_dates,
> > +                                            string='Gantt start date',
> > +                                            type='datetime',
> > +                                            multi='gantt_dates'),
> > +        'date_end_gantt': fields.function(_get_gantt_dates,
> > +                                          string='Gantt end date',
> > +                                          type='datetime',
> > +                                          multi='gantt_dates'),
> > +        'date_planned_end': fields.function(_get_date_end,
> > +                                            string='End Date',
> > +                                            type='datetime',
> > +                                            store=False),
> > +    }
>
> @jco-odoo <https://github.com/jco-odoo> Yes. IMHO you should respect the
> plan vs. actual. When was it planned to start and when did it actually
> start. When it was scheduled to end vs. when it was ended. Otherwise you
> will later totally loose the ability to review the performance (do we
> complete the MO's on time, or we are are always delayed)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OCA/manufacture/pull/28/files#r45977819>.
>